kasa/device.py

```diff
@@ -343,6 +345,18 @@ async def wifi_join(self, ssid: str, password: str, keytype: str = "wpa2_psk"):
     async def set_alias(self, alias: str):
         """Set the device name (alias)."""
 
+    @property
+    def descriptors(self) -> Dict[str, Descriptor]:
```

Comment:
So if we do call it Features we would break the API here (maybe that's why you figured on `Descriptors`).  Maybe we could create a class called `Features` that inherits from `set` or implements `__iter__` just with feature names and also implements `__getitem__` for the dictionary like behaviour.  Alternatively we could just make the deprecated `SmartDevice` implement the old features signature and issue a warning.

Reply:
I'm all for sunsetting and/or repurposing the current "features" property. It will be a breaking change, but I doubt it's ever been widely used directly. Only a very small subset of devices had it, and it was only for figure out if emeter is available.

Reply:
I don't think it's worth to complicate the code to keep the earlier API (could be doable by creating DummyFeatures for the legacy ones on IOT devices, but that would require filtering etc.

I have now repurposed the `features` to return a `Dict[str, Feature]`.